byoqm/source_repository/source_repository.py

```diff
@@ -8,7 +9,8 @@
 import chardet
 
 _TREESITTER_BUILD: Path = Path("build/my-languages.so")
-
+_IGNORE_FILE_PATH: Path = Path("byoqm/util/.ignore_paths.txt")
+_IGNORE_FILE_EXTENSIONS_PATH: Path = Path("byoqm/util/.ignore_extensions.txt")
```

Comment:
For configuration I think it will be very useful for the user to be able to specify the ignore file's path as an flag to the CLI, we can have a default but it should definitely be possible to override it.

I also think that the most reasonable default would be an ignore file in the directory from which the tool is being called, similar to how Git looks at the .gitignore file that is a sibling to the .git folder first.